File: unload-manager/src/main/java/com/starship/unloadmanager/config/ApplicationConfiguration.java
package com.starship.unloadmanager.config;

import com.starship.unloadmanager.client.SpaceMarineClient;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestTemplate;

@Configuration
public class ApplicationConfiguration {
    @Bean
    public RestTemplate restTemplate() {
        return new RestTemplate();
    }

    @Bean
    public SpaceMarineClient spaceMarineClient(@Value("${space.marine.url}") String spaceMarineUrl, RestTemplate restTemplate) {
        return new SpaceMarineClient(spaceMarineUrl, restTemplate);
    }
}

File: unload-manager/src/main/kotlin/com/starship/unloadmanager/config/ApplicationConfiguration.kt
package com.starship.unloadmanager.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.jndi.JndiObjectFactoryBean

@Configuration
class ApplicationConfiguration {
    @Bean
    fun ejbBean(): JndiObjectFactoryBean {
        val factoryBean = JndiObjectFactoryBean()
        factoryBean.jndiName = "ejb:/unload-manager-ejbs-0.0.1-jar-with-dependencies/SpaceMarineClientImpl!com.starship.unloadmanager.SpaceMarineClientInterface"
        factoryBean.isResourceRef = true
        return factoryBean
    }
}
